File: backend/src/main/java/com/cy/crm/module/followup/entity/FollowUp.java
package com.cy.crm.module.followup.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cy.crm.common.entity.AuditableEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_follow_up")
public class FollowUp extends AuditableEntity {

    private Long customerId;
    private Long projectId;
    private Long opportunityId;
    private String currentStage;
    private String nextStage;
    private String stageFeedback;
    private LocalDate followUpDate;
    private String followUpMethod;
    private Long contactId;
    private String content;
    private String nextPlan;
}

File: frontend/src/api/followup.ts
import { request } from './client'

export interface FollowUpPage {
  records?: FollowUpVO[]
  total?: number
  size?: number
  current?: number
}

export interface FollowUpVO {
  id?: number
  customerId?: number
  customerName?: string
  projectId?: number
  projectName?: string
  opportunityId?: number
  currentStage?: string
  currentStageName?: string
  nextStage?: string
  nextStageName?: string
  stageFeedback?: string
  followUpDate?: string
  followUpMethod?: string
  followUpMethodName?: string
  contactId?: number
  contactName?: string
  content?: string
  nextPlan?: string
  createdBy?: number
  createdByName?: string
  createdAt?: string
}

export interface FollowUpRequest {
  customerId: number
  projectId?: number
  opportunityId?: number
  currentStage?: string
  nextStage?: string
  stageFeedback?: string
  followUpDate: string
  followUpMethod?: string
  contactId?: number
  content: string
  nextPlan?: string
}

export function pageFollowUps(params: {
  customerId?: number
  current?: number
  size?: number
} = {}) {
  return request<FollowUpPage>({
    url: '/follow-ups',
    method: 'GET',
    params: {
      customerId: params.customerId,
      current: params.current ?? 1,
      size: params.size ?? 10,
    },
  })
}

export function createFollowUp(data: FollowUpRequest) {
  return request<number>({ url: '/follow-ups', method: 'POST', data })
}

export function updateFollowUp(id: number, data: FollowUpRequest) {
  return request<void>({ url: `/follow-ups/${id}`, method: 'PUT', data })
}

export function deleteFollowUp(id: number) {
  return request<void>({ url: `/follow-ups/${id}`, method: 'DELETE' })
}
